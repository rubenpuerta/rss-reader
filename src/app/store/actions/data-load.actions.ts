import { createAction, props, union } from '@ngrx/store';

import { NewsItem } from '@models/rss-news.model';

const initialDataLoad = createAction('[RSS] Initial data load');
const initialDataLoadSuccess = createAction(
	'[RSS] Initial data load success',
	props<{ news: NewsItem[] }>()
);
const initialDataLoadFailure = createAction('[RSS] Initial data load failure');

const actions = { initialDataLoad, initialDataLoadSuccess, initialDataLoadFailure };

const actionsUnions = union(actions);

export type DataLoadctionsType = typeof actionsUnions;
export const DataLoadActions = actions;
