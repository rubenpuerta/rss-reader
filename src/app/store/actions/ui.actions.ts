import { createAction, union } from '@ngrx/store';

const startLoading = createAction('[UI] Start loading');
const stopLoading = createAction('[UI] Stop loading');

const actions = { startLoading, stopLoading };

const actionsUnions = union(actions);

export type UIActionsType = typeof actionsUnions;
export const UIActions = actions;
