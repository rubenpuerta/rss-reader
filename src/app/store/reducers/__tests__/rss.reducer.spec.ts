import { NewsItem } from '@models/rss-news.model';
import { DataLoadActions } from '@store/actions/data-load.actions';
import { rssInitialState, rssReducer, RssState } from '@store/reducers/rss.reducer';

const mockedNews: NewsItem[] = [
	{
		author: 'test-name',
		description: 'test-html-code',
		guid: 'test-guid',
		pubDate: 12345,
		title: 'test-title',
		imageUrl: 'test-url-path-image',
		articleBrief: 'test-article-brief'
	}
];

describe('rssReducer', () => {
	let initialState: RssState;

	beforeEach(() => {
		initialState = { ...rssInitialState };
	});

	it('should contain initial state', () => {
		const result = rssReducer(undefined, { type: '' } as any);
		expect(result).toMatchSnapshot();
	});
	describe('DataLoadActions.initialDataLoadSuccess', () => {
		it('should set initial load of rss', () => {
			const result = rssReducer(
				initialState,
				DataLoadActions.initialDataLoadSuccess({ news: mockedNews })
			);
			expect(result).toMatchSnapshot();
		});
	});
});
