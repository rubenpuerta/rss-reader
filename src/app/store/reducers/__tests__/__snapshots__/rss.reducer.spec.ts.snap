// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rssReducer DataLoadActions.initialDataLoadSuccess should set initial load of rss 1`] = `
Object {
  "news": Array [
    Object {
      "articleBrief": "test-article-brief",
      "author": "test-name",
      "description": "test-html-code",
      "guid": "test-guid",
      "imageUrl": "test-url-path-image",
      "pubDate": 12345,
      "title": "test-title",
    },
  ],
}
`;

exports[`rssReducer should contain initial state 1`] = `
Object {
  "news": Array [],
}
`;
